### PR TITLE
add co function

### PR DESCRIPTION
**Pull request by benperez**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const params = Array(n + 1).join('x').replace(/x/g, function(_, n) {
+    return n === 0 ? '$0' : ', $' + n;
+  });
+  return Function(  // jshint ignore:line
+    'f',
+    'return function(' + params + ') { return f.apply(this, arguments); }'
+  ).call(this, f);
+};
+
+exports.co = function(generator) {
```
Comment (Bpless) [inline index.js:19]: Comments please.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const generator_length = generator.length;
+  return arity(n, () => {
```
Comment (Bpless) [inline index.js]: Should n be replaced with generator_length?

Comment (benperez) [inline index.js]: I clearly still need tests for these :zap:

Comment (davidchambers) [inline index.js]: … and linting. ;)

**Pull request merged**